### PR TITLE
[stas-molchanov]Axit tab switch and validation

### PR DESCRIPTION
**Pull request by moskdev**

### CODE REVIEW

**Comment by moskdev** [comment]
https://moskdev.github.io/ui-internship-lwo2/axit-interactive/

File: axit-interactive/index.html [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean
+          euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo.
+        </p>
+        <button class="btn-border">Download</button>
+      </div>
+      <form name="free-trial-form" class="signup-form mt-md" action="#">
+        <h3 class="signup-form--heading">Try Your <ins>free </ins>Trial Today</h3>
+        <fieldset id="free-trial-fs">
+          <input required type="text" title="Enter only latin letters" placeholder="Name" id="free-trial-name" onblur="validate('free-trial-name',/^[a-zA-Z ]+$/)"
+            onkeyup="validate('free-trial-name',/^[a-zA-Z ]+$/)">
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: Don't use inline events.
Move logic to JS. Check everywhere.

File: axit-interactive/index.html [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+          <input type="email" name="email" id="free-trial-email" placeholder="Email" onblur="validate('free-trial-email',/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/)"
+            onkeyup="validate('free-trial-email',/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/)">
+          <input type="password" title="Password should contain at least 1 lowercase letter,1 uppercase letter, 1 digit and 1 special symbol and be 6+ characters"
+            name="password" id="free-trial-password" placeholder="Password" onblur="validate('free-trial-password',/(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&+-]).{6,20}/)"
+            onkeyup="validate('free-trial-password',/(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&+-]).{6,20}/)">
+        </fieldset>
+        <input type="submit" value="Get Started" onclick="validateForm('free-trial-fs')">
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: Form has event submit.
Use it instead:
`object.addEventListener("submit", myScript);`

File: axit-interactive/index.html [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+      </ul>
+      <ul class="social--links-list__mb">
+        <li><a class="social-link__fb-sm" href="#" title="facebook"></a></li>
+        <li><a class="social-link__tw-sm" href="#" title="twitter"></a></li>
+        <li><a class="social-link__gp-sm" href="#" title="google+"></a></li>
+        <li><a class="social-link__pn-sm" href="#" title="pinterest"></a></li>
+        <li><a class="social-link__ins-sm" href="#" title="instagram"></a></li>
+        <li><a class="social-link__stu-sm" href="#" title="stubon upon"></a></li>
+        <li><a class="social-link__rss-sm" href="#" title="RSS"></a></li>
+      </ul>
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: DRY.
Use JS to change classes.

File: axit-interactive/index.html [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+          </p>
+        </div><img src="img/image_city_2.png" alt="Image of City 2">
+      </section>
+    </article>
+    <section class="about" id="about">
+      <h2 class="section-heading">why this is awesome</h2><span class="small-heading">Lorem ipsum dolor sit amet,
+        consectetur adipiscing elit.</span>
+      <div class="benefits">
+        <div class="benefits--block">
+          <figure></figure>
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: Why do you need this?

File: axit-interactive/index.html [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+              <li>100 websites</li>
+              <li>advanced customization</li>
+              <li>wordpress integration</li>
+              <li>24/7 customer support</li>
+            </ul>
+          </div>
+        </div>
+      </div>
+    </section>
+    <section class="reviews" id="reviews">
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: To be not confused use different names for classes and IDs.

File: axit-interactive/index.html [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+      <h2 class="section-heading">what our customers are saying</h2><span class="small-heading">Lorem ipsum dolor sit
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: Align please:
```
      <h2 class="section-heading">what our customers are saying</h2>
      <span class="small-heading">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
```
Check everywhere.

File: axit-interactive/index.html [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+          navMenu.classList.add('show');
+        } else {
+          navMenu.classList.remove('show');
+        }
+        // change navMenu state
+        menuShow = !menuShow;
+      });
+    })(document);
+
+  </script>
```
Comment (AlexLarusso) [inline axit-interactive/index.html]: Move to js file.

File: axit-interactive/js/script.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+  node.classList.remove('success');
+  node.classList.add('error');
+}
+
+function validateForm(id) {
+  const inputArr = Array.from(document.getElementById(id).children);
+  for (let i = 0; i < inputArr.length; i++) {
+    if (Array.from(inputArr[i].classList).indexOf('error') !== -1) {
+      alert('please enter valid values');
```
Comment (AlexLarusso) [inline axit-interactive/js/script.js]: Don't use alerts.

File: axit-interactive/js/script.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    return;
+  }
+
+  node.classList.remove('success');
+  node.classList.add('error');
+}
+
+function validateForm(id) {
+  const inputArr = Array.from(document.getElementById(id).children);
+  for (let i = 0; i < inputArr.length; i++) {
```
Comment (AlexLarusso) [inline axit-interactive/js/script.js]: empty line before.

File: axit-interactive/js/script.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    document.getElementById('features-para').innerText = featureParas[node.id.slice(-1)];
+    var myopacity = 0;
+    paraFade();
+
+    function paraFade() {
+      if (myopacity < 1) {
+        myopacity += .03;
+        setTimeout(function () {
+          paraFade()
+        }, 16);
```
Comment (AlexLarusso) [inline axit-interactive/js/script.js]: Why 16?
Why not 15 or 24?

File: axit-interactive/js/script.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    paraFade();
+
+    function paraFade() {
+      if (myopacity < 1) {
+        myopacity += .03;
+        setTimeout(function () {
+          paraFade();
+        }, 15);
+      }
+      document.getElementById('feature-content').style.opacity = myopacity;
```
Comment (AlyonaPianykh) [inline axit-interactive/js/script.js]: Can you create effect with opacity using css and not in code?

**Pull request merged**